cmake/kokkos_arch.cmake

```diff
@@ -1212,9 +1212,8 @@ if(KOKKOS_ENABLE_HIP AND NOT AMDGPU_ARCH_ALREADY_SPECIFIED AND NOT KOKKOS_IMPL_A
     )
   else()
     execute_process(COMMAND ${ROCM_ENUMERATOR} OUTPUT_VARIABLE GPU_ARCHS)
-    string(LENGTH "${GPU_ARCHS}" len_str)
-    # enumerator always output gfx000 as the first line
-    if(${len_str} LESS 8)
+    # Exits early if no GPU was detected
+    if("${GPU_ARCHS}" STREQUAL "")
```

Comment:
What happens with older versions of rocm if no GPU is detected with that change?

Reply:
it should fail at finding the gfx000 in the list of architectures

Reply:
Right, you get an error message that we could not automatically find a GPU that we support.